src/main/java/com/easypost/net/EasyPostResource.java

```diff
@@ -429,25 +435,38 @@ private static EasyPostResponse makeURLConnectionRequest(EasyPostResource.Reques
 			apiKey = EasyPost.apiKey;
 		}
 
-		String query;
-		try {
-			query = createQuery(params);
-		} catch (UnsupportedEncodingException e) {
-			throw new EasyPostException(
-                String.format("Unable to encode parameters to %s. Please email %s for assistance.", CHARSET, EasyPostResource.EASYPOST_SUPPORT_EMAIL), e);
+		String query = null;
+        JsonObject body = null;
+		if (params != null) {
+			switch (method) {
+				case GET:
+				case DELETE:
+					try {
+						query = createQuery(params);
+					} catch (UnsupportedEncodingException e) {
+						throw new EasyPostException(
+								String.format("Unable to encode parameters to %s. Please email %s for assistance.", CHARSET, EasyPostResource.EASYPOST_SUPPORT_EMAIL), e);
+					}
+					break;
+				case POST:
+				case PUT:
+					body = createBody(params);
```

Comment:
Should we add a try-catch block here?

Reply:
We could. The createBody() handles null values upstream, but for other errors that might arise, good idea.